libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -63,6 +51,33 @@ class MapboxDirectionsSession(
         })
     }
 
+    override fun requestFasterRoute(
+        adjustedRouteOptions: RouteOptions,
+        routesRequestCallback: RoutesRequestCallback
+    ) {
+        if (routes.isEmpty()) {
+            return
+        }
+        this.routeOptions = adjustedRouteOptions
```

Comment:
It seems we're changing the "global" route options when `requestFasterRoute` is called. Is that intentional? Could that be a problem?

Reply:
Thanks @Guardiola31337 
This was a very good catch. 

The question I would like to ask is do we want to change the global `routeOptions` if the user selects a faster route? 

It is different from `requestRoute` where user doesn't have the option to accept the route. This global `routeOptions` should only be changed if the user agrees to accept the faster route.

Reply:
Synced up internally and we're going to get the route options from the `DirectionsRoute`, set them when routes are set and remove the explicit set from `requestRoutes` and `requestFasterRoute`.